### PR TITLE
merge-client-and-client-context

### PR DESCRIPTION
**Pull request by qiaozha**
fix issue #1134

### CODE REVIEW

**Comment by qiaozha** [comment]
@ramya-rao-a @deyaaeldeen Could you help review this PR? Thanks

File: src/generators/clientFileGenerator.ts [hunk 1]
```diff
@@ -362,3 +436,135 @@ function writeClientOperations(
...
+  sourceFile.addStatements([
+    `\n\n`,
+    `const packageName = "${packageDetails.name || ""}";`,
+    `const packageVersion = "${packageDetails.version || ""}";`
+  ]);
+}
+
+function packageDetails(clientFile: SourceFile, packageDetails: any) {
+  throw new Error("Function not implemented.");
+}
```
Comment (deyaaeldeen) [inline src/generators/clientFileGenerator.ts:570]: I am not sure what is this but I guess it was added by mistake?
```suggestion
```

File: src/generators/clientFileGenerator.ts [hunk 2]
```diff
@@ -362,3 +436,135 @@ function writeClientOperations(
...
+    ...options,
+    userAgent: options.userAgent
+      ? \`\${options.userAgent} \${defaultUserAgent}\`
+      : \`\${defaultUserAgent}\`
+  });
+  
+  this.requestContentType = "application/json; charset=utf-8";
+  
+  `
+    : `// Initializing default values for options
```
Comment (deyaaeldeen) [inline src/generators/clientFileGenerator.ts:501]: NIT: Perhaps it will be more readable if each case is factored into its own function?

**Review by deyaaeldeen — APPROVED** [review]
The generated code looks good!

File: src/generators/clientFileGenerator.ts (hunk 2 above)
Comment (qiaozha) [reply src/generators/clientFileGenerator.ts:501]: Actually, I migrate them from ClientContextFileGenerator, but sure, let me change it.

**Comment by qiaozha** [comment]
merged it as all comments should have been resolved. @deyaaeldeen let me know if you have more comments. Thanks

**Pull request merged**